KicadModTree/PolygonPoints.py

```diff
@@ -0,0 +1,88 @@
+# KicadModTree is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# KicadModTree is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kicad-footprint-generator. If not, see < http://www.gnu.org/licenses/ >.
+#
+# (C) 2016-2018 by Thomas Pointhuber, <thomas.pointhuber@gmx.at>
+
+from KicadModTree.Point import Point2D
+from KicadModTree.nodes.Node import Node
+
+
+class PolygonPoints(object):
+    r"""Representation of multiple points for creating polygons
+
+    :Keyword Arguments:
+        * *nodes* (``list(Point)``) --
+          2D points describing the "polygon"
+        * *polygone* (``list(Point)``) --
+          alternative naming for the nodes parameter for backwards compatibility.
+        * *x_mirror* (``[int, float](mirror offset)``) --
+          mirror x direction around offset "point"
+        * *y_mirror* (``[int, float](mirror offset)``) --
+          mirror y direction around offset "point"
+
+    :Example:
+
+    >>> from KicadModTree import *
+    >>> PolyPoint([(0, 0),(1, 0)])
+    >>> PolyPoint([{'x': 0, 'y':0}, {'x': 1, 'y':0}])
+    """
+    def __init__(self, **kwargs):
+        self._initMirror(**kwargs)
+        self._initNodes(**kwargs)
+
+    def _initNodes(self, **kwargs):
+        self.nodes = []
+        if 'nodes' in kwargs:
+            for n in kwargs['nodes']:
+                self.nodes.append(Point2D(n))
+            if 'polygone' in kwargs:
+                raise KeyError('Use of "nodes" and "polygone" parameter at the same time is not supported.')
+        elif 'polygone' in kwargs:
```

Comment:
Add deprecation warning:

```python
warnings.warn(
            "polygone argument is deprecated, use nodes instead",
            DeprecationWarning
)
```

Reply:
Good idea. Added. (Will be fun running one of the larger generator scripts. There will be a lot of warnings displayed.)